enterprise/app/trends/trends_chart.tsx

```diff
@@ -38,34 +38,37 @@ interface Props {
   onSecondaryBarClicked?: (date: string) => void;
 }
 
-const TrendsChartTooltip = ({
-  active,
-  payload,
-  labelFormatter,
-  valueFormatter,
-  secondaryValueFormatter,
-  extractSecondaryValue,
-}: any) => {
-  if (active) {
+interface TrendsChartTooltipProps extends TooltipProps<any, any> {
+  labelFormatter: (datum: any) => string;
+  valueFormatter?: (datum: number) => string;
+  secondaryValueFormatter?: (datum: number) => string;
+  extractSecondaryValue?: (datum: any) => number;
+}
+
+class TrendsChartTooltip extends React.Component<TrendsChartTooltipProps> {
+  render() {
+    const p = this.props;
```

Comment:
uber nit: might want to write out `this.props.<blah>` to be consistent with the code style for the other components -- or alternatively make it a function component again and replace `any` with TrendsChartTooltipProps (then you could reference params directly instead of props)

Reply:
dunno what i was doing yesterday, somehow convinced myself that i couldn't make strict mode happy with a function component because of something about the way the tooltip worked, but obviously not the case.  swapped this back because i really didn't like have 100-character lines that were repeating `this.props` over and over.